components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java

```diff
@@ -98,8 +110,11 @@ public static NamedJdbcTemplate getNewTemplate() throws OrganizationManagementSe
 
         try {
             if (dataSource == null) {
-                dataSource = DatabaseUtil.getRealmDataSource(CarbonContext.getThreadLocalCarbonContext().getUserRealm()
-                        .getRealmConfiguration());
+                if (CarbonContext.getThreadLocalCarbonContext().getUserRealm() == null) {
+                    throw handleServerException(ERROR_CODE_ERROR_RETRIEVING_UM_DATASOURCE);
```

Comment:
We can use ```OrganizationManagementServerException handleServerException(
            OrganizationManagementConstants.ErrorMessages error, Throwable e, String... data) ``` method here by passing `null` object for throwable. Data is optional
```suggestion
                    throw handleServerException(ERROR_CODE_ERROR_RETRIEVING_UM_DATASOURCE, null);
```

Reply:
Fixed by - https://github.com/wso2-extensions/identity-organization-management-core/pull/9/commits/40966d139481cd6704211c3fccc7a8f0b2e69039